test/ResourceGeneratorTest.php

```diff
@@ -44,19 +45,28 @@ class ResourceGeneratorTest extends TestCase
 
     use ProphecyTrait;
 
-    /** @var ObjectProphecy|ServerRequestInterface */
+    /**
+     * @var ObjectProphecy|ServerRequestInterface
+     * @psalm-var ObjectProphecy<ServerRequestInterface>
+     */
```

Comment:
```suggestion
    /** @var ObjectProphecy&ServerRequestInterface */
```

Simpler, and more consistent with what we do in other packages.